inc/css/blocks/class-button-group-css.php

```diff
@@ -157,7 +157,7 @@ public function render_css( $block ) {
 						'property' => 'line-height',
 						'value'    => 'lineHeight',
 						'format'   => function( $value, $attrs ) {
-							return is_numeric( $value ) ? $value . 'px' : $value;
+							return ! is_string( $value ) && is_numeric( $value ) ? $value . 'px' : $value;
```

Comment:
Did you test it with legacy blocks to confirm this works as well with everything?

Reply:
By legacy, I will refer to the first version, which used the `int` as values. Those will be the same since they will be pixels. 

The more recent version, which uses the unit input field, is now fixed when using the unitless mode.